README.md

```diff
@@ -12,33 +12,50 @@ npm install sd-jwt-vc
 
 ### Issuer
 
-This is a TypeScript class that represents an issuer of Verifiable Credentials (VCs) that can create Signed and Disclosed JWTs (SD JWTs) for VCs. It uses the sd-jwt library to create the SD JWTs.
+This is a TypeScript class that represents an issuer of Verifiable Credentials (VCs) that can create Signed and Disclosed JWTs (SD JWTs) for VCs.
 
 #### Usage
 
-To use the Issuer class, you need to create an instance of it by passing in a signer configuration object and a hasher configuration object. Here's an example:
+To use the Issuer class, you need to create an instance of it by passing in a signer configuration object and a hasher configuration object.
+signer configuration callback function will be used to sign the SD JWTs and hasher configuration callback function will be used to hash the disclosued claims in the SD JWTs.
+
+Here's an example:
 
 ```typescript
-import { generateKeyPair } from 'jose';
+import { DisclosureFrame, Hasher, SDJWTPayload, Signer, base64encode } from '@meeco/sd-jwt';
+import { createHash } from 'crypto';
+import { JWTHeaderParameters, JWTPayload, KeyLike, SignJWT, exportJWK, generateKeyPair } from 'jose';
+import { HasherConfig, Issuer, SignerConfig, VCClaims, defaultHashAlgorithm, supportedAlgorithm } from '../src/index';
+
+const hasherCallbackFn = function (alg: string = defaultHashAlgorithm): Hasher {
+  return (data: string): string => {
+    const digest = createHash(alg).update(data).digest();
+    return base64encode(digest);
+  };
+};
 
-import { DisclosureFrame, SDJWTPayload } from '@meeco/sd-jwt';
-import { Issuer } from './issuer';
-import { signerCallbackFn } from './test-utils/helpers';
-import { HasherConfig, SignerConfig, VCClaims } from './types';
-import { hasherCallbackFn, supportedAlgorithm } from './util';
+const signerCallbackFn = function (privateKey: Uint8Array | KeyLike): Signer {
+  return (protectedHeader: JWTHeaderParameters, payload: JWTPayload): Promise<string> => {
+    return new SignJWT(payload).setProtectedHeader(protectedHeader).sign(privateKey);
+  };
+};
 
-const keyPair = await generateKeyPair(supportedAlgorithm.EdDSA);
+async function main() {
+  const keyPair = await generateKeyPair(supportedAlgorithm.EdDSA);
 
-let hasher: HasherConfig = {
-  alg: 'sha256',
-  callback: hasherCallbackFn('sha256'),
-};
-let signer: SignerConfig = {
-  alg: supportedAlgorithm.EdDSA,
-  callback: signerCallbackFn(keyPair.privateKey),
-};
+  const hasher: HasherConfig = {
+    alg: 'sha256',
```

Comment:
@vijayshiyani @ragnika is this value later used anywhere else?
I see that spec is using a little different notation to specify the alg `sha-256`.

Reply:
both of them are valid.  In JavaScript, both notations can be used interchangeably.

`openssl list -digest-algorithms`

> The `algorithm` is dependent on the available algorithms supported by the * version of OpenSSL on the platform. Examples are `'sha256'`, `'sha512'`, etc.
> On recent releases of OpenSSL, `openssl list -digest-algorithms` will
> display the available digest algorithms.

Reply:
Then maybe, let's stick to the one used in sd-jwt specification? one including `-`?
It is as well used in @meeco/sd-jwt library.